iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/E4KConnectionMessagePayload.java

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.azure.sdk.iot.device.transport.mqtt;
+
+import com.google.gson.annotations.Expose;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * The payload object to be sent in all connect/disconnect messages sent to the E4K MQTT broker. Not to be confused
+ * with the payload of the MQTT CONNECT/DISCONNECT packet.
+ */
+@Builder
+public class E4KConnectionMessagePayload
+{
+    /**
+     * The connection state to report. For instance, after connecting to the MQTT broker, this SDK sends a message
+     * with {@link E4KConnectionState#Connected}. Before disconnecting, this SDK sends a message with
+     * {@link E4KConnectionState#Disconnected}.
+     */
+    @Expose(deserialize = false)
+    @Getter
+    @Setter
+    private E4KConnectionState connectionState;
+
+    /**
+     * The user agent string for this client.
+     */
+    @Expose(deserialize = false)
+    @Getter
+    @Setter
+    private String deviceClientType;
+
+    /**
+     * The version of MQTT that this client uses to connect to the MQTT broker.
+     */
+    @Expose(deserialize = false)
+    @Getter
+    @Setter
+    private String mqttVersion;
+
+    /**
+     * The optional PnP model Id for this client.
+     */
+    @Expose(deserialize = false)
+    @Getter
+    @Setter
+    private String modelId;
```

Comment:
Does model Id appear only with the context of PnP? If not, how about we rename this to "pnpModelId"?

Reply:
We should leave it to match the property name we have in the SDK as well as the value that was sent in the connect packet.

Reply:
I'll leave this as-is for now then